pdu/Src/tempSensor.c

```diff
@@ -22,15 +24,28 @@
 /*********************************************************************************************************************/
 /*------------------------------------------------------Macros-------------------------------------------------------*/
 /*********************************************************************************************************************/
-
+#define SENSOR_ADDRESS_READ 0x91
+#define SENSOR_ADDRESS_WRITE 0x90
+#define CONFIG_REGISTER 0x01
+#define TEMP_REGISTER 0x00
 /*********************************************************************************************************************/
 /*-------------------------------------------------Global variables--------------------------------------------------*/
 /*********************************************************************************************************************/
-
+uint8_t tempregister_info [2];
+float temperature_reading;
+uint8_t CONFIG_VALUE = 0x28;
 /*********************************************************************************************************************/
 /*------------------------------------------------Function Definition------------------------------------------------*/
 /*********************************************************************************************************************/
-
+float temperature_conversion(uint8_t temp_array[2]){
+    uint16_t temp = (temp_array[0] << 4) | (temp_array[1] >> 4);
```

Comment:
I would add a comment to explain what you are doing here so that when other developers read it, they can understand what you are doing without referencing the datasheet which is time consuming.